app/src/main/java/br/org/cesar/knot_setup_app/activity/login/LoginPresenter.java

```diff
@@ -51,14 +52,63 @@ public void doLogin(String password){
         this::onErrorHandler);
     }
 
+    private void getOpenthreadConfig(){
+        String bearer,ip;
+        bearer = dataManager.getInstance()
+                .getPersistentPreference()
+                .getSharedPreferenceString(context,"token");
+
+        ip = dataManager.getInstance()
+                .getPreference()
+                .getSharedPreferenceString(context,"ip");
+
+        this.port = dataManager.getInstance()
+                .getPreference().getSharedPreferenceString(context,"port");
+
+        this.request = "http://" + ip + ":" + port +"/api/radio/openthread";
+
+        dataManager.getInstance().getService().getOpenthreadConfig(this.request,bearer)
+                .timeout(30, TimeUnit.SECONDS)
+                .subscribeOn(Schedulers.newThread())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(this::getOpenThreadSucceeded,
+                        this::onErrorHandler);
+
+    }
+
+    private void getOpenThreadSucceeded(Openthread openthread){
+
+        dataManager.getInstance().getPreference()
+                .setSharedPreferenceString(context,"netname",openthread.getNetworkName());
+
+        dataManager.getInstance().getPreference()
+                .setSharedPreferenceString(context,"channel",openthread.getChannel());
+
+        dataManager.getInstance().getPreference()
+                .setSharedPreferenceString(context,"xpanid",openthread.getXpanId());
+
+        dataManager.getInstance().getPreference()
+                .setSharedPreferenceString(context,"panid", openthread.getPanId());
+
+        dataManager.getInstance().getPreference()
+                .setSharedPreferenceString(context,"ipv6",openthread.getMeshIpv6());
+
+        dataManager.getInstance().getPreference()
+                .setSharedPreferenceString(context,"masterkey",openthread.getMasterKey());
+
+        mViewModel.callbackOnLogin();
+    }
+
+
+
     private void loginSucceeded(User user) {
         dataManager.getInstance().getPersistentPreference()
                 .setSharedPreferenceString(context,"email",email);
 
         dataManager.getInstance().getPersistentPreference()
                 .setSharedPreferenceString(context,"token","Bearer " + user.getToken());
 
-        mViewModel.callbackOnLogin();
+        getOpenthreadConfig();
```

Comment:
Are you using this anywhere else?
If not, please, remove it

Reply:
Will do when getting rid of leftovers.

Reply:
This patch is supposed to be self-contained. Please, remove it or refer to a PR that removes it.